CONTRIBUTING.md

```diff
@@ -18,6 +18,105 @@ Every pull request should document all changes made in the [changelog].
 If your name does not already appear in the [AUTHORS] file, please feel free to
 add it as part of your patch.
 
+## Preparing for a new release
+
+If you're already contributing to this project and want to do more,
+then there might be a chance to help out with the preparation of new releases.
+Whether you're new or have prepared druid releases many times already,
+it helps to follow a checklist of what needs to be done. This is that list.
+
+### Changelog
+
+- Add a new *Unreleased* section by copying the current one.
+  Keep the sections, but delete the entries.
+- Rename the old *Unreleased* section to the target release version and add the release date.
+- Add the correct link for the target release revision to the bottom of the file.
+- Delete any empty sections.
+- Tidy up the entries, possibly reordering some for more logical grouping.
+
+### Dependencies
+
+**We only test and specify the newest versions of dependencies.** Read on for more details.
+
+Rust dependencies like druid specify their own sub-dependencies in `Cargo.toml`.
+These specifications are usually version ranges according to [semver],
+stating that the dependency requires a sub-dependency of the specified version
+or any newer version that is still compatible. It is up to the final application
+to choose which actual versions get used via the `Cargo.lock` file of that application.
+
+Because the final application chooses the sub-dependency versions and they are most likely
+going to be higher than the minimum that is specified in our `Cargo.toml` file,
+we need to make sure that druid works properly with these newer versions.
+Yes according to [semver] rules they should work, but library authors make mistakes
```

Comment:
I am inclined to trust reported semver versions, and also trust maintainers of crates we use (and those should generally be few crates, of a high standard) to yank releases that break semver.